src/main/resources/native/android/dalvik/source/BleService.java

```diff
@@ -0,0 +1,214 @@
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.helloandroid;
+
+import android.app.Activity;
+import static android.app.Activity.RESULT_OK;
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.le.BluetoothLeScanner;
+import android.bluetooth.le.ScanCallback;
+import android.bluetooth.le.ScanRecord;
+import android.bluetooth.le.ScanResult;
+
+import android.content.Intent;
+
+import java.util.Arrays;
+import java.util.Formatter;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class BleService  {
+
+    private final MainActivity activity;
+    private static final Logger LOG = Logger.getLogger(BleService.class.getName());
+    private BluetoothLeScanner scanner;
+
+    private final static int REQUEST_ENABLE_BT = 1001;
+    private final List<String> uuids = new LinkedList<>();
+
+
+    public BleService(MainActivity a) {
+System.err.println("BLEService constructed");
+        this.activity = a;
+        init();
+System.err.println("BLEService constructed done, start scanning now");
+        startScanning();
+System.err.println("BLEService constructed done111");
+    }
+
+    private void init() {
+System.err.println("init 0");
+        BluetoothAdapter adapter = BluetoothAdapter.getDefaultAdapter();
+System.err.println("init 1");
+        if (!adapter.isEnabled()) {
+System.err.println("init 2");
+            Intent enableBtIntent = new Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE);
+            IntentHandler intentHandler = new IntentHandler() {
+                @Override
+                public void gotActivityResult (int requestCode, int resultCode, Intent intent) {
+System.err.println("init 3");
+                    if (requestCode == REQUEST_ENABLE_BT && resultCode == RESULT_OK) {
+                        scanner = adapter.getBluetoothLeScanner();
+                    }
+                }
+            };
+
+System.err.println("init 4");
+            if (activity == null) {
+                LOG.log(Level.WARNING, "Activity not found. This service is not allowed when "
+                        + "running in background mode or from wearable");
+                return;
+            }
+System.err.println("init 5");
+            activity.setOnActivityResultHandler(intentHandler);
+
+            // A dialog will appear requesting user permission to enable Bluetooth
+System.err.println("init 6");
+            activity.startActivityForResult(enableBtIntent, REQUEST_ENABLE_BT);
+System.err.println("init 7");
+
+        } else {
+System.err.println("init 8");
+            scanner = adapter.getBluetoothLeScanner();
+System.err.println("init 9");
+        }
+
+    }
+
+    private ScanCallback scanCallback;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=5148382)